### PR TITLE
editor: select with options tree

### PR DESCRIPTION
**Pull request by sebastiendeleze**
This new component replaces the default `select` type.

* Adds a custom select field type, to allow the hierachization of options.
* Removes useless `SelectWithSortComponent` and points `selectWithSort` type to `select`.
* Removes fallback to `selectWithSort` type in editor.
* Removes translations of options, as it is done by the new component.
* Contributes to make rero/sonar#530 work.

Co-Authored-by: Sébastien Délèze <sebastien.deleze@rero.ch>

### CODE REVIEW

File: projects/rero/ng-core/src/lib/record/editor/type/select-tree/select-tree.component.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
+/*
+ * RERO angular core
+ * Copyright (C) 2020 RERO
```
Comment (Garfield-fr) [inline projects/rero/ng-core/src/lib/record/editor/type/select-tree/select-tree.component.ts:3]: `2021`

**Review by Garfield-fr — APPROVED** [review]

File: projects/rero/ng-core/src/lib/record/editor/type/custom-select/custom-select.component.html [hunk 2]
```diff
@@ -0,0 +1,44 @@
+<!--
+ RERO angular core
+ Copyright (C) 2020 RERO
```
Comment (Garfield-fr) [inline projects/rero/ng-core/src/lib/record/editor/type/custom-select/custom-select.component.html:3]: `2021`

File: projects/rero/ng-core/src/lib/record/editor/type/custom-select/custom-select.component.scss [hunk 3]
```diff
@@ -0,0 +1,36 @@
+/*
+ * RERO angular core
+ * Copyright (C) 2020 RERO
```
Comment (Garfield-fr) [inline projects/rero/ng-core/src/lib/record/editor/type/custom-select/custom-select.component.scss:3]: `2021`

File: projects/rero/ng-core/src/lib/record/editor/type/custom-select/custom-select.component.scss [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+.btn-group > button {
+  color: #495057 !important;
```
Comment (Garfield-fr) [inline projects/rero/ng-core/src/lib/record/editor/type/custom-select/custom-select.component.scss:18]: Put the color in a bootstrap variable

File: projects/rero/ng-core/src/lib/record/editor/type/custom-select/custom-select.component.scss [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  font-size: 0.9rem;
+  border: 1px solid #ced4da !important;
```
Comment (Garfield-fr) [inline projects/rero/ng-core/src/lib/record/editor/type/custom-select/custom-select.component.scss:20]: Put the color in a bootstrap variable

File: projects/rero/ng-core/src/lib/record/editor/type/custom-select/custom-select.component.ts [hunk 6]
```diff
@@ -0,0 +1,222 @@
+/*
+ * RERO angular core
+ * Copyright (C) 2020 RERO
```
Comment (Garfield-fr) [inline projects/rero/ng-core/src/lib/record/editor/type/custom-select/custom-select.component.ts:3]: `2021`

**Review by Garfield-fr — APPROVED** [review]

**Review by zannkukai — APPROVED** [review]

**Pull request merged**